pkg/controllers/util/propagatedversion.go

```diff
@@ -20,138 +20,138 @@ are Copyright 2023 The KubeAdmiral Authors.
-		if currentMaxSurge, ok2, err := unstructured.NestedString(clusterObj.Object, MaxSurgePathSlice...); err == nil &&
-			ok == ok2 {
-			if desiredMaxSurge == currentMaxSurge {
-				needUpdate = false
-			}
-		}
-	} else if desiredMaxSurge, ok, err := unstructured.NestedInt64(desiredObj.Object, MaxSurgePathSlice...); err == nil {
-		if currentMaxSurge, ok2, err := unstructured.NestedInt64(clusterObj.Object, MaxSurgePathSlice...); err == nil && ok == ok2 {
-			if desiredMaxSurge == currentMaxSurge {
-				needUpdate = false
-			}
-		}
-	}
-	if needUpdate {
-		return true
-	}
-
-	needUpdate = true
-	if desiredMaxUnavailable, ok, err := unstructured.NestedString(desiredObj.Object, MaxUnavailablePathSlice...); err == nil {
-		if currentMaxUnavailable, ok2, err := unstructured.NestedString(clusterObj.Object, MaxUnavailablePathSlice...); err == nil &&
-			ok == ok2 {
-			if desiredMaxUnavailable == currentMaxUnavailable {
-				needUpdate = false
-			}
-		}
-	} else if desiredMaxUnavailable, ok, err := unstructured.NestedInt64(desiredObj.Object, MaxUnavailablePathSlice...); err == nil {
-		if currentMaxUnavailable, ok2, err := unstructured.NestedInt64(clusterObj.Object, MaxUnavailablePathSlice...); err == nil && ok == ok2 {
-			if desiredMaxUnavailable == currentMaxUnavailable {
-				needUpdate = false
-			}
-		}
-	}
-	if needUpdate {
-		return true
-	}
-	// If versions match and the version is sourced from the
-	// generation field, a further check of metadata equivalency is
-	// required.
-	return strings.HasPrefix(targetVersion, generationPrefix) && !ObjectMetaObjEquivalent(desiredObj, clusterObj)
-}
-
-// SortClusterVersions ASCII sorts the given cluster versions slice
-// based on cluster name.
-func SortClusterVersions(versions []fedcorev1a1.ClusterObjectVersion) {
-	sort.Slice(versions, func(i, j int) bool {
-		return versions[i].ClusterName < versions[j].ClusterName
-	})
-}
-
-// PropagatedVersionStatusEquivalent returns true if both statuses are equal by
-// comparing Template and Override version, and their ClusterVersion slices;
-// false otherwise.
-func PropagatedVersionStatusEquivalent(pvs1, pvs2 *fedcorev1a1.PropagatedVersionStatus) bool {
-	return pvs1.TemplateVersion == pvs2.TemplateVersion &&
-		pvs1.OverrideVersion == pvs2.OverrideVersion &&
-		reflect.DeepEqual(pvs1.ClusterVersions, pvs2.ClusterVersions)
-}
-
-func ConvertVersionMapToGenerationMap(versionMap map[string]string) map[string]int64 {
-	generationMap := make(map[string]int64, len(versionMap))
-	for key, version := range versionMap {
-		if strings.HasPrefix(version, resourceVersionPrefix) {
-			generationMap[key] = 0
-			continue
-		}
-		if !strings.HasPrefix(version, generationPrefix) {
-			continue
-		}
-
-		generationString := strings.TrimPrefix(version, generationPrefix)
-		generation, err := strconv.ParseInt(generationString, 10, 64)
-		if err != nil {
-			continue
-		}
-		generationMap[key] = generation
-	}
-	return generationMap
-}
+// import (
```

Comment:
Why do we comment out this file? The target branch is already uncompilable without commenting out this file.